test/Microsoft.AspNet.Mvc.Core.Test/Formatters/OutputFormatterTests.cs

```diff
@@ -101,6 +102,48 @@ public static IEnumerable<object[]> SelectResponseCharacterEncodingData
             Assert.Equal(new StringSegment(expectedContentType), formatterContext.ContentType);
         }
 
+        [Fact]        
+        public void WriteResponse_AddOrOverrideCharsetReturnsMediaTypeFromCache_IfEncodingIsUtf8()
+        {
+            // Arrange
+            var formatter = new Mock<OutputFormatter>();
+
+            formatter
+                .Setup(f => f.SelectCharacterEncoding(It.IsAny<OutputFormatterWriteContext>()))
+                .Returns(Encoding.UTF8);
+
+            var mediaTypeCache = new Dictionary<string, string>
+            {
+                {
+                    "application/json",
+                    "application/json; charset=utf-8"
+                }
+            };
+
+            formatter
+                .Protected()
+                .SetupGet<IDictionary<string, string>>("OutputMediaTypeCache")
+                .Returns(mediaTypeCache);
```

Comment:
You can write the same tests without complicated mockery by just verifying that it returns the same string twice.